pkg/auth/providers/azure/clients/conversions.go

```diff
@@ -15,52 +15,56 @@ import (
 
 const Name = "azuread"
 
-// UserToPrincipal attempts to convert a raw user from Graph SDK to a Principal.
-// The user may be of type returned by Azure AD Graph API (deprecated and to be removed soon)
-// or Microsoft Graph API.
-func UserToPrincipal(user interface{}) (v3.Principal, error) {
-	var id, displayName, loginName string
+func (c azureADGraphClient) userToPrincipal(user interface{}) (v3.Principal, error) {
+	u, ok := user.(graphrbac.User)
+	if !ok {
+		return v3.Principal{}, errors.New("wrong type of user, needs to be a valid Azure AD Graph user")
+	}
+	return v3.Principal{
+		ObjectMeta:    metav1.ObjectMeta{Name: Name + "_user://" + *u.ObjectID},
+		DisplayName:   *u.DisplayName,
+		LoginName:     *u.UserPrincipalName,
+		PrincipalType: "user",
+		Provider:      Name,
+	}, nil
+}
 
-	switch u := user.(type) {
-	case msgraph.User:
-		id = *u.ID
-		displayName = *u.DisplayName
-		loginName = *u.UserPrincipalName
-	case graphrbac.User:
-		id = *u.ObjectID
-		displayName = *u.DisplayName
-		loginName = *u.UserPrincipalName
-	default:
-		return v3.Principal{}, errors.New("wrong type of user, needs to be a valid Azure AD Graph user or Microsoft Graph user")
+func (c azureADGraphClient) groupToPrincipal(group interface{}) (v3.Principal, error) {
+	g, ok := group.(graphrbac.ADGroup)
+	if !ok {
+		return v3.Principal{}, errors.New("wrong type of group, needs to be a valid Azure AD Graph group")
 	}
+	p := v3.Principal{
+		ObjectMeta:    metav1.ObjectMeta{Name: Name + "_group://" + *g.ObjectID},
+		DisplayName:   *g.DisplayName,
+		PrincipalType: "group",
+		Provider:      Name,
+	}
+	return p, nil
+}
 
+func (c azureMSGraphClient) userToPrincipal(user interface{}) (v3.Principal, error) {
```

Comment:
Nit:

I would recommend moving these functions to their respective client files and deleting the conversions file. There's no real reason to mix methods for different structs in this file.